### PR TITLE
Fix moderation but in which empty field was throwing an error

### PR DESCRIPTION
**Pull request by eob**

### CODE REVIEW

File: src/agents/onboarding_agent.py [hunk 1]
```diff
@@ -26,12 +26,20 @@
 
 
 def _is_allowed_by_moderation(user_input: str, openai_api_key: str) -> bool:
-    start = time.perf_counter()
```
Comment (eob) [inline src/agents/onboarding_agent.py:29]: None -> Permitted

File: src/agents/onboarding_agent.py [hunk 2]
```diff
@@ -26,12 +26,20 @@
 
 
 def _is_allowed_by_moderation(user_input: str, openai_api_key: str) -> bool:
```
Comment (eob) [inline src/agents/onboarding_agent.py:28]: Change is to include:

- empty input
- open ai failure

as "permitted" responses

File: src/agents/onboarding_agent.py [hunk 3]
```diff
@@ -210,6 +218,7 @@ def run(self, context: AgentContext) -> Action:  # noqa: C901
             )
             game_state.chat_history_for_onboarding_complete = True
 
+        game_state.onboarding_agent_has_completed = True
```
Comment (eob) [inline src/agents/onboarding_agent.py:221]: Adds an explicit "Agent has completed its work" flag

File: src/endpoints/onboarding_endpoints.py [hunk 4]
```diff
@@ -92,13 +92,30 @@ def set_character_description(
...
+    def complete_onboarding(self, **kwargs) -> bool:  # noqa: C901
         """Attempts to complete onboarding."""
         start = time.perf_counter()
         try:
             context = self.agent_service.build_default_context()
             game_state = get_game_state(context)
 
+            # These fields are required for the game to continue. If we're calling complete_onboarding and they're not
+            # set, we should fail early.
+            if not game_state.player.name:
```
Comment (eob) [inline src/endpoints/onboarding_endpoints.py:104]: fail hard and early if these fields are empty -- they shouldn't be

File: src/schema/game_state.py [hunk 5]
```diff
@@ -79,6 +79,10 @@ class GameState(BaseModel):
         description="The key of the last question asked to the user via context_utils.await_ask.",
     )
 
+    onboarding_agent_has_completed: Optional[bool] = Field(
```
Comment (eob) [inline src/schema/game_state.py:82]: be more explicit about whether the onboarding agent has run. before we were sort of duck-typing it (see below)

File: src/schema/game_state.py [hunk 6]
```diff
@@ -113,6 +117,7 @@ def is_onboarding_complete(self) -> bool:
             self.player is not None
             and self.player.is_onboarding_complete()
             and self.chat_history_for_onboarding_complete
+            and self.onboarding_agent_has_completed
```
Comment (dkolas) [inline src/schema/game_state.py:120]: suggestion (non-blocking): If we have the explicit flag, do we want to just check that exclusively?  ANDing the others seems to only leave the door open for new strange cases.

File: src/endpoints/onboarding_endpoints.py (hunk 4 above)
Comment (dkolas) [reply src/endpoints/onboarding_endpoints.py:104]: 👍

**Review by dkolas — APPROVED** [review]
LGTM - one optional suggestion below

File: src/schema/game_state.py (hunk 6 above)
Comment (eob) [reply src/schema/game_state.py:120]: Yeah I think that feels right - i'll make a note to add a cleanup pr for it

**Pull request merged**